### ISSUE TITLE
Context is caching old values in db after ExecuteScalar

### ISSUE THREAD

**Issue by johanskoldekrans** [dotnet/efcore#6391]
### Steps to reproduce

First I read data from the database on an entity that has foreign keys at a values-table into my model.

``` C#
model.FormSubmission = ctx.FormSubmissions
    .Where(fs => fs.CustomerId == id && fs.ProcessedDate == null && fs.CancelDate == null && fs.FormId == form.FormId)
    .Include(fs => fs.FormSubmissionState)
    .Include(fs => fs.FormSubmissionValues) //here should be old values
    .Include(f => f.FormSubmissionChapters).ThenInclude(fc => fc.FormChapter)
    .FirstOrDefault();
```

Then I execute a command using ExecuteScalar updating the values table.

``` C#
var command = ctx.Database.GetDbConnection().CreateCommand();
command.CommandType = CommandType.StoredProcedure;
command.CommandText = "sp_FormSubmissionValidate";
var param = command.CreateParameter();
param.ParameterName = "@FormID";
param.Value = form.FormId;
command.Parameters.Add(param);
command.Connection.Open();
command.ExecuteScalar();
command.Connection.Close();
```

If I read the values-table after the procedure has been run I get the old values from the database and not the ones that got updated with the procedure I ran in the ExecuteScalar.

``` C#
model.FormSubmissionValues = ctx.FormSubmissionValues
    .Where(fs => fs.FormSubmissionId == model.FormSubmission.FormSubmissionId)
    .ToList();
```

I understand that a caching mechanism occurs since i touch the same entities (which I will remove - saw this by mistake) but how can I make sure it reads from the database and not from cache? sometimes it is nice to be sure to be sure...

EF Core version: 1.1.0-alpha-21937
Operating system: Win10
Visual Studio version: VS 2015 Update 3

**Comment by ErikEJ** [comment]
You are going down to the ADO.NET level, and this is not connected with EF change tracking. You need to re-read the updated data using LINQ.

**Comment by johanskoldekrans** [comment]
Is there a way to exec the procedure using EF as of now (I now it is not implemented yet)? It is not returning anything. I guess I could make it return FormSubmissionValues and use .FromSQL instead but I really want to avoid it if possible.

By Linq you mean using "from x where"-instead?

**Comment by johanskoldekrans** [comment]
I tried these two but with the same result (cached values):
`model.FormSubmissionValues = ctx.FormSubmissionValues.FromSql("Select * from dbo.FormSubmissionValues Where FormSubmissionId = {0}", model.FormSubmission.FormSubmissionId).ToList();`
                `model.FormSubmissionValues = (from fs in ctx.FormSubmissionValues where fs.FormSubmissionId == model.FormSubmission.FormSubmissionId select fs).ToList();`

**Comment by ajcvickers** [comment]
@johanskoldekrans I think what you are seeing is that when the context is already tracking an entity, then any query for that entity will not bring back a new entity or update any values in the entity, but rather just return the same instance that is being tracked.

This usually only causes problems when the context is being kept around for more than a single unit of work, so the usual recommendation here is to use a short lived context such that the query that happens after the update is running with a new context instance.

Another way to deal with this is to use Reload, but that hasn't been implemented yet--it is being tracked by issue #1203. You can usually do the equivalent of reload by making an AsNoTracking query for the entity, which will always return the values from the database, and then copying these values into the entity and setting its state to Unchanged.

**Comment by johanskoldekrans** [comment]
I was hoping for a reload-option or a way to force the entity to acknowledge the change. I can easily workaround it for now. Thanks for the info and help! Would be nice to force a reload by stating changed objects by the stored procedure.
